packages/eslint-plugin/lib/rules/prefer-icon-component.js

```diff
@@ -0,0 +1,70 @@
+const jsxAstUtils = require('jsx-ast-utils');
+
+const { elementType, getProp, getLiteralPropValue } = jsxAstUtils;
+
+const MESSAGE =
+  'The <va-icon> Web Component should be used instead of Font Awesome. See: https://design.va.gov/storybook/?path=/docs/uswds-va-icon--default';
```

Comment:
Should we link to the migration docs instead (to be written)?

Reply:
This might be a good place to add icon migration steps as part of [this ticket's work](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2536) ➡️  https://design.va.gov/about/developers/using-web-components#how-to-migrate-to-web-components

Reply:
Once this PR is merged, I'll update the link to point there: 
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/pull/2620